core/src/apps/base.py

```diff
@@ -183,6 +183,64 @@ async def handle_DoPreauthorized(
     return await handler(ctx, req, authorization.get())  # type: ignore [Expected 2 positional arguments]
 
 
+async def handle_UnlockPath(ctx: wire.Context, msg: UnlockPath) -> protobuf.MessageType:
+    from trezor.crypto import hmac
+    from trezor.messages import Cancel, UnlockedPathRequest
+    from trezor.ui.layouts import confirm_action
+    from apps.common.paths import SLIP25_PURPOSE
+    from apps.common.seed import Slip21Node, get_seed
+    from apps.common.writers import write_uint32_le
+
+    _KEYCHAIN_MAC_KEY_PATH = [b"TREZOR", b"Keychain MAC key"]
+
+    # UnlockPath is relevant only for SLIP-25 paths.
+    # Note: Currently we only allow unlocking the entire SLIP-25 purpose subtree instead of
+    # per-coin or per-account unlocking in order to avoid UI complexity.
+    if len(msg.address_n) != 1 or msg.address_n[0] != SLIP25_PURPOSE:
```

Comment:
`if msg.address_n != [SLIP25_PURPOSE]` would be cleaner

Reply:
https://github.com/trezor/trezor-firmware/pull/2289/commits/aafaadabaedac2e0f35e98a07985fb782bbd19b0